pkg/util/utils.go

```diff
@@ -343,7 +380,7 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx context.Context, k8sO
 // this function is used on the resources with VPC scenario, and the provided obj is the K8s CR which is
 // used to generate the NSX resource.
 func GenerateIDByObject(obj metav1.Object) string {
-	return normalizeNameByLimit(obj.GetName(), string(obj.GetUID()), common.MaxIdLength)
+	return normalizeNameByLimit(obj.GetName(), string(obj.GetUID()), common.MaxIdLength, nil)
```

Comment:
LGTM+1
Please take care of resource id change

Reply:
The new function `GenerateIDByObject` is called for the resources only with VPC scenario. Since VPC is the greenfield, so we don't need to think about the upgrade case.